docs/syz_verifier.md

```diff
@@ -0,0 +1,81 @@
+# syz-verifier
+
+Logical bugs are a type of  bugs that make a program operate incorrectly, possibly
+producing unintended output, but usually without causing the program to crash or
+producing assertion failures. Since programs with logical bugs generally compile,
+these are hard to detect and require more than dynamic analysis tools.
+
+Differential fuzzing aims to detect such bugs by providing the same input to different
+implementations (whose specifications say they behave identically) and cross-compares
+their resulting behaviour to determine whether its the same.
+
+`syz-verifier` is a differential fuzzing tool that cross-compares the execution of programs,
+generated using `syzlang`, on different versions of the Linux kernel.
+
+The architecture of `syz-verifier` is shown in the following diagram.
+
+
+![Architecture overview](syz_verifier_structure.png)
+
+The `syz-verifier` process wiil start and manage VM instances created from the kernels to be cross-compared. It will also start the `syz-runner` process on the VMs. Communication between the host and the guest will be done via RPC calls.
+
+`syz-verifier` will generate and send a continuous stream of programa to `syz-runner` via RPC  while `syz-runner` is responsible for starting `syz-executor` processes and turning the program into input for those. `syz-executor` will process the input, which triggers a sequence of syscalls in the kernel. Then, `syz-runner` collects the results and sends them back to the host.
+
+ When `syz-verifier` has received results from all the kernels for a specific program, it verifies them to ensure they are identical. If a mismatch is found, the program will be rerun on all the kernels to ensure the mismatch is not flaky (i.e. it didn't occur because of some background activity or external state). If the mismatch occurs in all reruns, `syz-verifier` will create a report for the program and write it to persistent storage.
+
+
+# How to use `syz-verifier`
+
+The tool is not currently integrated with `syzbot` so it needs to be run manually.
```

Comment:
I would either remove this sentence or move it to the bottom of the "how to use" section as a note.
Reader currently don't expect it to be integrated, so what they will look here for is just instructions on how to use it. Also even if it's integrated, the integration won't cover billions of combinations of all kernels, including closed forks maintained within some companies. So just describe how to use it.

Reply:
This still holds.